pkg/types/aws/validation/machinepool.go

```diff
@@ -1,14 +1,23 @@
 package validation
 
 import (
+	"fmt"
+	"strings"
+
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	"github.com/openshift/installer/pkg/types/aws"
 )
 
 // ValidateMachinePool checks that the specified machine pool is valid.
-func ValidateMachinePool(p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
+func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	for i, zone := range p.Zones {
+		if !strings.HasPrefix(zone, platform.Region) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("zones").Child(fmt.Sprintf("%d", i)), zone, fmt.Sprintf("Zone not in configured region (%s)", platform.Region)))
```

Comment:
Use `Index(i)` to reference the element of the list, instead of `Child(fmt.Sprintf(...))`

Reply:
> Use `Index(i)` to reference the element of the list, instead of `Child(fmt.Sprintf(...))`

Done with 00d5501e7 -> f25961d4b.